code/modules/events/grubinfestation_vr.dm

```diff
@@ -1,26 +1,35 @@
 /datum/event/grub_infestation
-	announceWhen	= 90
-	var/spawncount = 1
+	announceWhen = 10
+	endWhen = 11
+	var/location
+	var/locstring
 
+/datum/event/grub_infestation/start()
 
-/datum/event/grub_infestation/setup()
-	announceWhen = rand(announceWhen, announceWhen + 60)
-	spawncount = rand(4 * severity, 6 * severity)	//grub larva only have a 50% chance to grow big and strong
-	sent_spiders_to_station = 0
+	location = rand(0,8)
+	var/list/turf/simulated/floor/turfs = list()
+	var/spawn_area_type
+	spawn_area_type = /area/gateway/prep_room
+	locstring = "the solar farm"
 
-/datum/event/grub_infestation/announce()
-	command_announcement.Announce("Unidentified lifesigns detected coming aboard [station_name()]. Secure any exterior access, including ducting and ventilation.", "Lifesign Alert", new_sound = 'sound/AI/aliens.ogg')
+	for(var/areapath in typesof(spawn_area_type))
+		var/area/A = locate(areapath)
+		for(var/turf/simulated/floor/F in A.contents)
+			if(turf_clear(F))
+				turfs += F
 
+	var/list/spawn_types = list()
+	spawn_types = list(/mob/living/simple_animal/retaliate/solargrub)
+	var/max_number = 8
 
-/datum/event/grub_infestation/start()
-	var/list/vents = list()
-	for(var/obj/machinery/atmospherics/unary/vent_pump/temp_vent in world)
-		if(!temp_vent.welded && temp_vent.network && temp_vent.loc.z in using_map.station_levels)
-			if(temp_vent.network.normal_members.len > 50)
-				vents += temp_vent
+	spawn(0)
+		var/num = rand(2,max_number)
+		while(turfs.len > 0 && num > 0)
+			var/turf/simulated/floor/T = pick(turfs)
+			turfs.Remove(T)
+			num--
+			var/spawn_type = pick(spawn_types)
+			new spawn_type(T)
 
-	while((spawncount >= 1) && vents.len)
-		var/obj/vent = pick(vents)
-		new /obj/effect/spider/spiderling/grub(vent.loc)
-		vents -= vent
-		spawncount--
+/datum/event/grub_infestation/announce()
+	command_announcement.Announce("Bioscans indicate that solar grubs have somehow entered through the gateway. Clear them out, before this starts to affect the powergrid.", "Vermin infestation")

```

Comment:
The gateway? Do they only ever spawn in the gateway?

Reply:
Yup, any other areas you want them to spawn in? people complained about them ending up all over the station, even in like telecoms

Reply:
I say let them show up anywhere except some blacklisted areas, but _not_ use vents.

Or conversely, whitelisted areas, as defined in the original infestation code for mice and lizards.

Reply:
Actually am going to gather opinions on this via a vote.

Reply:
https://strawpoll.com/5c8ry1s7

Reply:
"All over station, excluding the dorms" has huge majority compared to other stuff.

Reply:
Alrite, i'll close this pull for now and go work on a fix so they can still spawn everywhere, just not the dorms.